src/layer/vector/Canvas.js

```diff
@@ -138,6 +138,18 @@ export var Canvas = Renderer.extend({
 	_initPath: function (layer) {
 		this._updateDashArray(layer);
 		this._layers[Util.stamp(layer)] = layer;
+		if (layer.options.keyboard) {
+			layer._path = DomUtil.create('div', 'leaflet-canvas-interactive', this._container);
+			layer._path.setAttribute('tabindex', 0);
+			layer._path.setAttribute('role', 'graphics-symbol img');
+			layer._path.setAttribute('aria-label', layer.options.title);
+			layer._path.setAttribute('title', layer.options.description);
```

Comment:
I notice the canvas `<title>` isn't producing a tooltip on hover (or focus, as is also the case in some browsers).

Reply:
Canvas title support is hardly possible at the moment. The unrelated example you mentioned uses a technique similar to what Leaflet does under the hood to show `Tooltip`s on canvas - calculate what related "figures" are under the mouse current position. I do not think implementing this for a native tooltip is necessary as we already have a `Tooltip` to achieve that.